feat(get-started): getting started and deploy your own app

KathrynLewis opened the pull request:
Resolves Jira: [CDAAS-2491]
Deploy preview https://deploy-preview-29--cdaas-docs.netlify.app/get-started/deploy-your-app/

[CDAAS-2491]: https://armory.atlassian.net/browse/CDAAS-2491?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Commit:
feat(get-started): getting started and deploy your own app

Commit:
link to rna overview until install page is in main

parth-srivastav commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -103,150 +114,71 @@ strategies:
-</tr>
-</table>
 
+In your CDaaS config, you can create deployment strategies with as many steps as you want. In this example, the deployment config defines the following strategies:
```
```suggestion
In your CD-as-a-Service config, you can create deployment strategies with as many steps as you want. In this example, the deployment config defines the following strategies:
```

parth-srivastav commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -10,50 +10,61 @@ tags: ["Deployment", "Quickstart"]
-* `trafficSplit`: 75% to the current version and 25% to the new version (prod)
 
-In addition, prod deployment requires a manual approval to begin deployment and another to continue deployment after the traffic split.
+Next, save the following config as `deployment.yaml` to the root of your app directory, next to the `manifests` directory. Replace the bracketed values with your own.
```
I think we need to guide users how to fill the bracketed values in apart from the comments in YAML. we need to make a distinction if the values are free-from logical names or if they need to match something precisely.

e.g. account needs to match the agentIdentifier, but target name is free form.

aimeeu replied:
To follow up on Parth comment, for the namespace files, we could tell the user what to name the namespaces. For example in the staging card code:

```
apiVersion: v1
kind: Namespace
metadata:
  name: staging
```

(benefit of hindsight. I wrote this page in haste and parth approved it in haste lol)

parth-srivastav commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -103,150 +114,71 @@ strategies:
 
-## Deploy your app
+## 2. Deploy your app
+In order to see some deployment strategies at work, you'll need to deploy your app twice. The first time, CDaaS will skip any traffic splitting or manual approval steps in order to have a stable version fully deployed. The second deploy will demonstrate an advanced deployment strategy, routing only 25% of traffic to your new version, while routing the rest to the stable version.
```
change CDaaS to CD-as-a-Service

parth-srivastav commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -103,150 +114,71 @@ strategies:
+The prod deployment requires a manual approval to begin deployment and another to continue deployment after the traffic split.
 
-## Deploy your app
+## 2. Deploy your app
```
Its probably worthwhile repeating in this section that - You need to have Armory CLI installed to start a deployment with CD-as-a-Service.

parth-srivastav commented on content/en/get-started/deploy-your-app.md:
````diff
@@ -10,50 +10,61 @@ tags: ["Deployment", "Quickstart"]
 ```
-<my-app>
+<your-app>
 ├── deployment.yaml
````
We should add a comment explaining that we are going to create this file here in this step.

parth-srivastav commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -103,150 +114,71 @@ strategies:
 
 ## {{% heading "nextSteps" %}}
 
+* {{< linkWithTitle "deployment/strategies/overview.md" >}}
```
I think we'll benefit from adding a custom text here but link to the same pages. 

e.g. 
Learn about different deployment strategies
Use Github Action to automatically trigger deployments
Integrate your tool chain using webhooks.

parth-srivastav commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -10,50 +10,61 @@ tags: ["Deployment", "Quickstart"]
 ## Learning objectives
 
-In this guide, you build on what you learned in the  CD-as-a-Service [Quickstart]({{< ref "get-started/quickstart" >}}) by deploying your own app using the deployment configuration from the Quickstart.
+1. [Create a deployment config file](#create-a-deployment-config-file)
```
these internal links don't auto scroll to the section.

parth-srivastav commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -10,50 +10,61 @@ tags: ["Deployment", "Quickstart"]
-In this guide, you build on what you learned in the  CD-as-a-Service [Quickstart]({{< ref "get-started/quickstart" >}}) by deploying your own app using the deployment configuration from the Quickstart.
+1. [Create a deployment config file](#create-a-deployment-config-file)
+2. [Deploy your app](#deploy-your-app)
+3. [Monitor your deployment](#monitor-your-deployment)
```
There is no bigger Monitor subsection Section to link to. We do teach users to click the link CLI generated.

KathrynLewis committed:
Update content/en/get-started/deploy-your-app.md

Co-authored-by: parth-srivastav <87822636+parth-srivastav@users.noreply.github.com>

aimeeu commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -10,50 +10,61 @@ tags: ["Deployment", "Quickstart"]
-Create your staging namespace in `namespace-staging.yaml` and your prod namespace in `namespace-prod.yaml`.  The deployment config file supports deploying manifests to specific targets, and the deployment config file uses separate namespace manifests to illustrate this feature.
 
-## Create your deployment config file
+## 1. Create a deployment config file
```
We don't need to put a number in the heading. The steps are in order on the page and listed numerically in the Learning Objectives section, so that's usually enough. 
Also, removing the number will fix the link in the Learning Objectives section

```suggestion
## Create a deployment config file
```

KathrynLewis replied:
This was updated 👍

aimeeu commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -10,50 +10,61 @@ tags: ["Deployment", "Quickstart"]
 
 ## Learning objectives
 
```
Add some kind of introductory statement

```suggestion
In this guide, you use a blue/green strategy to deploy your own app to your Kubernetes cluster. If you don't have a cluster, you can use [kind](https://kind.sigs.k8s.io/) to install Kubernetes locally. 

>If you prefer a web-based, interactive tutorial, see the CD-as-a-Service Console's [**Deploy Your Own App** tutorial](https://next.console.cloud.armory.io/getting-started).
```

aimeeu commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -10,50 +10,61 @@ tags: ["Deployment", "Quickstart"]
 ### Directory structure
 
-This guide assumes the following directory structure:
+In this guide you will create a deployment config and two simple namespace configs. The namespace manifests should be in a `manifests` directory along with the Kubernetes manifests for deploying your app.
```
```suggestion
In this guide you create a deployment config and two simple namespace configs. The namespace manifests should be in a `manifests` directory along with the Kubernetes manifests for deploying your app.
```

Removed "will" - [Use present tense](https://docs.armory.io/contribute/create/guides/docs-style-guide/#use-present-tense)

parth-srivastav commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -10,50 +10,61 @@ tags: ["Deployment", "Quickstart"]
 ## {{% heading "prereq" %}}
 
-* You have completed the CD-as-a-Service [Quickstart]({{< ref "get-started/quickstart" >}}).
+* You have the [Armory CLI installed]({{< ref "cli.md" >}}).
```
The page should stand on its own, Lets have collapsible section on how to install CLI, and how to install RNA.

See example here - https://github.com/armory/docs/pull/2047/files#diff-0f0bc3c26ed0b12cf6ab4fb2ddeade37fae2b10eb328d3ecce75cea1da27c17eR61

aimeeu commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -10,50 +10,61 @@ tags: ["Deployment", "Quickstart"]
 
-* You have completed the CD-as-a-Service [Quickstart]({{< ref "get-started/quickstart" >}}).
+* You have the [Armory CLI installed]({{< ref "cli.md" >}}).
+* You have an [Armory RNA installed]({{< ref "remote-network-agent/overview.md" >}}) in your cluster.
```
[Use active voice](https://docs.armory.io/contribute/create/guides/docs-style-guide/#use-active-voice)

Parth and I talked a while ago about putting an expandable "show me how" under each of these bullet points but that's not clear in the Jira ticket.

- You have installed the Armory CLI
-- Show me how expandable section with the install command
- You have installed a Remote Network Agent installed in your cluster
-- Show me how expandable section with the CLI command

(formatting not guaranteed to have survived copy/paste)
```
* You have installed the [Armory CLI]({{< ref "cli.md" >}}).

  <details><summary>Show me how</summary>

  </details>

* You have installed a [Remote Network Agent]({{< ref "remote-network-agent/overview.md" >}}) in your cluster.

  <details><summary>Show me how</summary>
  
  Mac/Linux:

  ```bash (you might have to use HTML tags instead of backticks)
  curl -sL go.armory.io/get-cli | bash
  ```

  Windows:

  Download the latest `armory-cli` Windows executable from the [repo Releases page](https://github.com/armory-io/armory-cli/releases). Install on the machine that has access to your Kubernetes cluster.

  </details>
```


```

Commit:
Merge remote-tracking branch 'origin/master' into kl/deploy-own-app-CDAAS-2491

aimeeu commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -10,50 +10,61 @@ tags: ["Deployment", "Quickstart"]
-## Create your deployment config file
+## 1. Create a deployment config file
+
+First create two manifests for the staging and prod namespaces. These are where you'll deploy your app and also showcases the ability to deploy manifests to specific targets. Save these to the `manifests` directory.
```
```suggestion
To illustrate deploying to different targets, you deploy to two namespaces in the same cluster. In production, you would deploy to different clusters. 

First, create two manifests, one for staging (`namespace-staging.yaml`) and the other for prod (`namespace-prod.yaml`). Save these to the `manifests` directory.
```

Use present tense

aimeeu commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -74,9 +85,9 @@ targets:
-  - path: manifests/your-app.yaml # replace with the name of your app manifest
-  - path: manifests/your-app-service.yaml # replace with the name of your app service manifest
-  - path: manifests/namespace-staging.yaml # 
+  - path: manifests/<your-app>.yaml # replace with the name of your app manifest
+  - path: manifests/<your-app-service>.yaml # replace with the name of your app service manifest
```
good catch on the placeholders!

aimeeu commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -103,150 +114,71 @@ strategies:
 
-## Deploy your app
+## 2. Deploy your app
+In order to see some deployment strategies at work, you'll need to deploy your app twice. The first time, CDaaS will skip any traffic splitting or manual approval steps in order to have a stable version fully deployed. The second deploy will demonstrate an advanced deployment strategy, routing only 25% of traffic to your new version, while routing the rest to the stable version.
```
Use present tense https://docs.armory.io/contribute/create/guides/docs-style-guide/#use-present-tense

aimeeu replied:
This is pretty important, so how about calling it out with an `alert` shortcode or  '>' ('>In order to...)

aimeeu commented on content/en/get-started/deploy-your-app.md:
````diff
@@ -103,150 +114,71 @@ strategies:
    ```
 
-   The `deploy start` command starts your deployment. The command returns a **Deployment ID** and a link to your deployment details. 
+   This command starts your deployment, then returns a **Deployment ID** and a link to your deployment details. 
````
```suggestion
   This command starts your deployment, and then returns a **Deployment ID** and a link to your deployment details. 
```

aimeeu commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -103,150 +114,71 @@ strategies:
+   > You must issue manual approvals using the UI. You cannot issue manual approvals using the CLI.
 
-   Since you defined a manual approval to deploy to prod in your strategy, you need to issue your manual approval using the UI. Once the `staging` deployment has completed, click **Approve** to allow the `prod` deployment to begin. Because this is the first time deploying your app, CD-as-a-Service deploys 100% to your prod environment, skipping the defined `trafficSplit` strategy. CD-as-a-Service ...
+  Because this is the first time deploying your app, CD-as-a-Service deploys 100% to your prod environment, skipping the defined `trafficSplit` strategy. CD-as-a-Service uses the `trafficSplit` when deploying subsequent versions of your app.
```
Is this supposed to be part of #4? If yes, I think you need to add another space for the sentence to align.

![image](https://github.com/armory/docs-cdaas/assets/26799583/a75e89a3-455d-4516-9d86-b4c59b27f41f)

aimeeu commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -103,150 +114,71 @@ strategies:
+{{% alert title="Important" color="warning" %}}
+CDaaS manages your Kubernetes deployments using ReplicaSets. During the initial deployment of your app, CDaaS deletes the underlying Kubernetes deployment object in a way that leaves behind the ReplicaSet and pods so that there is no actual downtime for your app. These are later deleted when the deployment succeeds.
+
+If your initial deployment fails, you should [manually delete]({{< ref "troubleshooting/tools#initial-deployment-failure-orphaned-replicaset" >}}) the orphaned ReplicaSet.
```
```suggestion
CD-as-a-Service manages your Kubernetes deployments using ReplicaSets. During the initial deployment of your app, CD-as-a-Service  deletes the underlying Kubernetes deployment object in a way that leaves behind the ReplicaSet and pods so that there is no actual downtime for your app. These are later deleted when the deployment succeeds.

If your initial deployment fails, you should [manually delete]({{< ref "troubleshooting/tools#initial-deployment-failure-orphaned-replicaset" >}}) the orphaned ReplicaSet.
```
Can't use CDaaS as the product name. Has to be CD-as-a-Service, Armory CD-as-a-Service, or Armory Continuous Deployment-as-a-Service  https://docs.armory.io/contribute/create/guides/docs-style-guide/#armory-products-list

aimeeu commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -103,150 +114,71 @@ strategies:
 
 1. Update your Kubernetes manifest to deploy a new version of your app. 
-1. Start the deployment from the root of your directory.
+2. Start the deployment from the root of your directory.
```
Numercial lists should all be:
```
1.
1.
1.
```

The list is compiled to the correct numbers. Using all 1s saves you from having to renumber if you insert another step into the top or middle of the list.

aimeeu commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -103,150 +114,71 @@ strategies:
 
-1. Start the deployment from the root of your directory.
+2. Start the deployment from the root of your directory.
 
```
Numercial lists should all be:
```
1.
1.
1.
```

The list is compiled to the correct numbers. Using all 1s saves you from having to renumber if you insert another step into the top or middle of the list.

aimeeu commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -103,150 +114,71 @@ strategies:
-This `armory deploy status` command returns a point-in-time status and exits. It does not watch the deployment.
-
->You must issue manual approvals using the UI. You cannot issue manual approvals using the CLI.
+   Approve the deployment to prod again and this time, observe the traffic split on the deployment details page and preview your web app deployment. Issue your manual approval to finish deployment.
```
```suggestion
   Approve the deployment to prod and this time, observe the traffic split on the deployment details page. You can also preview your web app deployment. Issue your manual approval to finish deployment.
```
nits

aimeeu commented on content/en/get-started/deploy-your-app.md:
```diff
@@ -10,50 +10,61 @@ tags: ["Deployment", "Quickstart"]
->If you prefer a web-based, interactive tutorial, see the CD-as-a-Service Console's [**Deploy Your Own App** tutorial](https://next.console.cloud.armory.io/getting-started).
 
 ## {{% heading "prereq" %}}
 
```
I think we should include a point:

```
* You are familiar with the material in the [Quickstart]({{< ref "get-started/quickstart" >}}.
```

If we don't, then we should not assume the user knows deployment file syntax, how to issue manual approvals, or where to find the deploy preview link.

aimeeu commented on content/en/deployment/monitor-deployment.md:
```diff
@@ -9,35 +9,82 @@ tags: ["Monitor"]
 ---
 
 
```
Thanks for updating this page!

Commit:
PR feedback

aimeeu commented on content/en/get-started/overview.md:
```diff
@@ -2,10 +2,16 @@
 categories: ["Get Started", "Concepts"]
 tags: ["Quickstart", "Deployment"]
 ---
 
```
This is a great start. I think this page could use cardpanes/cards with different user journeys...

Also what paths to take if you don't have a K8s cluster -> use the UI sandbox route, use kind to install locally...

more thinking needed for future iterations

aimeeu reviewed (changes requested):
Great first draft!

KathrynLewis committed:
Update content/en/get-started/deploy-your-app.md

Co-authored-by: aimee <aimeeu.opensource@gmail.com>

KathrynLewis committed:
Update content/en/get-started/deploy-your-app.md

Co-authored-by: aimee <aimeeu.opensource@gmail.com>

KathrynLewis committed:
Update content/en/get-started/deploy-your-app.md

Co-authored-by: aimee <aimeeu.opensource@gmail.com>

KathrynLewis committed:
Update content/en/get-started/deploy-your-app.md

Co-authored-by: aimee <aimeeu.opensource@gmail.com>

KathrynLewis committed:
Update content/en/get-started/deploy-your-app.md

Co-authored-by: aimee <aimeeu.opensource@gmail.com>

aimeeu committed:
add more to deploy your own app;

add include file for install cli tabpane
update relevant pages to use include file

aimeeu committed:
add ## What you need to use CD-as-a-Service

aimeeu committed:
replace tabpane with include file

aimeeu commented:
After some thought, I made a few changes so that this tutorial includes the Quickstart sections Sign up for CD-as-a-Service, Install the CD-as-a-Service CLI, and Connect your cluster.   This based on the decision not to assume/require the user do the Quickstart. Also pulled info from other parts of the quickstart and included in this to explain concepts or flesh out steps.

Commit:
linting
